apps/ejabberd/src/http_upload/http_upload_s3.erl

```diff
@@ -0,0 +1,144 @@
+%%==============================================================================
+%% Copyright 2016 Erlang Solutions Ltd.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%==============================================================================
+
+-module(http_upload_s3).
```

Comment:
I still don't like the inconsistency. If we don't like the naming convention for mods backends, then I think it should be an atomic, global change. If S3 upload feature will be reused later by some other extension, then I would have nothing against creating common `http_upload_s3` module, while retaining `mod_http_upload_s3` backend module that only forwards the call to `http_upload_s3`.